frontend/src/containers/RegisterPage/RegisterPage.js

```diff
@@ -1,31 +1,75 @@
 import { Grid } from '@material-ui/core';
 import React from 'react';
+import { register } from '../../api/PublicAPI';
+import CustomSnackbar from '../../components/CustomSnackbar/CustomSnackbar';
 import RegistrationForm from '../../components/RegistrationForm/RegistrationForm';
 
 class RegisterPage extends React.Component {
 
-  // [TM]: TODO WDB-28 also delete line below
-  // eslint-disable-next-line no-unused-vars 
+  constructor(props) {
+    super(props);
+    this.state = {
+      showGeneralError: false,
+      showUserExistsError: false
+    };
+  }
+
   handleRegister = (firstName, lastName, email, password) => {
-    // Call to BE
-    // After successful registration reroute to confirm email
-    const { history } = this.props;
-    history.push('/confirm-email');
+    const data = {
+      firstName,
+      lastName,
+      email,
+      password
+    };
+    register(data).then(() => {
+      const { history } = this.props;
+      history.push('/confirm-email');
+    })
+      .catch(ex => {
+        const { status } = ex.response;
+        if (status === 400) this.setState({ showGeneralError: true });
```

Comment:
Maybe it would nice to also include 500 responses here, just in case something unexpected happend at backend. In this case, this `if` could be converted to `else`, while `else if` below could become `if`

Reply:
done